### PR TITLE
Add additional warn-unused flags

### PR DESCRIPTION
**Pull request by tgodzik**
Added new flags:
- `-Ywarn-unused:privates`
- `-Ywarn-unused:patvars` - one small issue, but easily fixable
- `-Ywarn-unused:locals`
- `-Ywarn-unused:implicits`

I couldn't add `-Ywarn-unused:explicits` because it didn't work well with inheritance. I enabled it for a while and removed anything that seemed no longer needed.

### CODE REVIEW

**Comment by kpbochenek** [comment]
```
[info] Main Scala API documentation to /home/runner/work/metals/metals/mtags/target/scala-2.11.12/api...
[info] Non-compiled module 'compiler-bridge_2.11' for Scala 2.11.12. Compiling...
[info] Compiling 125 Scala sources to /home/runner/work/metals/metals/mtags/target/scala-2.11.12/classes ...
[info]   Compilation completed in 13.755s.
[error] 'privates' is not a valid choice for '-Ywarn-unused'
[error] bad option: '-Ywarn-unused:privates'
[error] 'privates' is not a valid choice for '-Ywarn-unused'
[error] bad option: '-Ywarn-unused:privates'
[error] (mtags / Compile / compileIncremental) Compilation failed
[error] (mtags / Compile / doc) Scaladoc generation failed
```

**Comment by tgodzik** [comment]
> ```
> [info] Main Scala API documentation to /home/runner/work/metals/metals/mtags/target/scala-2.11.12/api...
> [info] Non-compiled module 'compiler-bridge_2.11' for Scala 2.11.12. Compiling...
> [info] Compiling 125 Scala sources to /home/runner/work/metals/metals/mtags/target/scala-2.11.12/classes ...
> [info]   Compilation completed in 13.755s.
> [error] 'privates' is not a valid choice for '-Ywarn-unused'
> [error] bad option: '-Ywarn-unused:privates'
> [error] 'privates' is not a valid choice for '-Ywarn-unused'
> [error] bad option: '-Ywarn-unused:privates'
> [error] (mtags / Compile / compileIncremental) Compilation failed
> [error] (mtags / Compile / doc) Scaladoc generation failed
> ```

Should be fixed now.

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 1]
```diff
@@ -1851,7 +1849,6 @@ class MetalsLanguageServer(
       check()
       buildTools
         .loadSupported()
-        .foreach(_.onBuildTargets(workspace, buildTargets))
```
Comment (ckipp01) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:1854]: I'm confused by this, what was the `onBuildTargets` method here supposed to be doing?

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:1854]: It's another leftover from Pants, came up when I added the explicits flag.

File: .scalafix.conf [hunk 2]
```diff
@@ -1,10 +1,13 @@
 rules = [
   OrganizeImports,
   ExplicitResultTypes,
+  RemoveUnused
```
Comment (tgodzik) [inline .scalafix.conf:4]: Added back the `RemoveUnused` rule to remove unused private and local values. Left imports to the OrganizeImports rules.

**Pull request merged**